code/game/machinery/bots/cleanbot.dm

```diff
@@ -271,6 +271,15 @@ text("<A href='?src=\ref[src];operation=oddbutton'>[src.oddbutton ? "Yes" : "No"
 			armed++
 			icon_state = "roombot_battle[on]"
 			icon_initial = "roombot_battle"
+	else if (istype(W, /obj/item/clothing/head/maidhat))
+		var/obj/machinery/bot/cleanbot/roomba/meido/M = new(get_turf(src))
+		M.name = name
+		if(integratedpai)
```

Comment:
```suggestion
		M.install_pai(eject_integrated_pai_if_present())
```
This should also correctly transfer the verb. I suggesting testing this

Reply:
Seems to work fine